server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -170,15 +170,27 @@ final private[worker] class StorageManager(conf: RssConf, workerSource: Abstract
   private var fileInfosDb: DB = null
   // ShuffleClient can fetch data from a restarted worker only
   // when the worker's fetching port is stable.
-  if (RssConf.workerGracefulShutdown(conf)) {
-    try {
-      val recoverFile = new File(RssConf.workerRecoverPath(conf), RECOVERY_FILE_INFOS_FILE_NAME)
-      this.fileInfosDb = LevelDBProvider.initLevelDB(recoverFile, CURRENT_VERSION)
-      reloadAndCleanFileInfos(this.fileInfosDb)
-    } catch {
-      case e: Exception =>
-        logError("Init level DB failed:", e)
-        this.fileInfosDb = null
+
+  {
+    if (RssConf.workerGracefulShutdown(conf)) {
+      try {
+        val recoverFile = new File(RssConf.workerRecoverPath(conf), RECOVERY_FILE_INFOS_FILE_NAME)
+        this.fileInfosDb = LevelDBProvider.initLevelDB(recoverFile, CURRENT_VERSION)
+        reloadAndCleanFileInfos(this.fileInfosDb)
+      } catch {
+        case e: Exception =>
+          logError("Init level DB failed:", e)
+          this.fileInfosDb = null
+      }
+    }
+    cleanupExpiredAppDirs(System.currentTimeMillis())
```

Comment:
I just realized that we should avoid deleting shuffle data restored from leveldb, which is in fileInfos' keySet.